### PR TITLE
added accessibility props for touchables

### PR DESCRIPTION
**Pull request by fangmobile**
accessibilityLabels are missing in these touchable*.js files.
for #5322 
@ide This is not tested yet. I will update with test.

### CODE REVIEW

**Comment by mkonicek** [comment]
lgtm

cc @sahrens

**Comment by fangmobile** [comment]
my coworker @annjose verified this fix works.
Basically she build react native from source off my fork, added accibilityLabel in touchable highlight in UI explorer example like so
````
<TouchableHighlight
            accessibilityLabel="AnnAccessLabel_TouchableHighlight_WithImage"
            style={styles.wrapper}
            onPress={() => console.log('stock THW image - highlight')}>
            <Image
              source={heartImage}
              style={styles.image}
            />
````
And verified the the label shows up in uiautomator viewer.
Please see attached image
Before this fix the content description will be empty even if you set the accessibilityLabel in code.
@ide @mkonicek @sahrens Please let us know whether this is good enough?

**Comment by fangmobile** [comment]
As we was working on another task in our project we noticed textinput does not have accessibility because it is using a touchable. This is more serious than touchable. In touchables sometimes you can get the child view's accessibility label, such as the "Press Me" string in TcouchaExample.js. In textinput.js there is just touchable. And if it is not accessible then the text input field is totally not accessible. 

I update this PR with a fix in textinput.js. Attaching some "before" and "after" screenshots. This is tested for Both iOS and Android. BTW the previous commit was also tested both in iOS and Android but there was not screen shot for iOS.
iOS before the fix. There is only "value", no accessibility label. 
screen shot 2016-01-27 at 2 58 09 pm

Android before the fix.
screen shot 2016-01-27 at 2 57 13 pm

ios after the fix
screen shot 2016-01-28 at 12 47 59 am


Android after the fix
screen shot 2016-01-28 at 12 40 07 am

File: Libraries/Components/TextInput/TextInput.js [hunk 1]
```diff
@@ -287,6 +287,27 @@ var TextInput = React.createClass({
...
+    accessibilityTraits: React.PropTypes.oneOfType([
+          React.PropTypes.oneOf(View.AccessibilityTraits),
+          React.PropTypes.arrayOf(React.PropTypes.oneOf(View.AccessibilityTraits)),
+        ]),
+    /**
+     * Used to set the accessibility component type of this view
+     * @platform android
+     */
+    accessibilityComponentType: React.PropTypes.oneOf(View.AccessibilityComponentType),
+    /**
```
Comment (mkonicek) [inline Libraries/Components/TextInput/TextInput.js]: All of these should be already pulled in via the `...View.propTypes` above (see `View.js`).

**Comment by mkonicek** [comment]
Besides the inline comment looks good. Thanks for adding the screenshots!

**Comment by fangmobile** [comment]
Thanks @mkonicek for catching this. 
Yes they are not needed. I put them in by mistake because I saw testID there but the rest missing. So when I remove them I also removed testId which should also come from View.js.
There is the quick test. Which the accessibility props still set in the example:
1. Remove them from the propTypes part but keep them in return function. Still works.
2. Remove them from both locations. Accessibility Label is gone.

This show they are not needed in propTypes but need to be returned.

Also squshed commits. Please let me know if there is anything I should do.

File: Examples/UIExplorer/TextInputExample.android.js [hunk 2]
```diff
@@ -208,7 +208,11 @@ exports.examples = [
   {
     title: 'Auto-focus',
     render: function() {
-      return <TextInput autoFocus={true} style={styles.singleLine} />;
+      return <TextInput
```
Comment (ide) [inline Examples/UIExplorer/TextInputExample.android.js]: code style, please write this as:
```js
return (
  <TextInput
    props.....
  />
);
```

**Comment by ide** [comment]
@fangmobile thanks for the test plan!

**Comment by fangmobile** [comment]
@ide Thanks. Amended another commit.

**Comment by fangmobile** [comment]
builds were failing due to an unrelated commit on master that was reverted. Now all good.
@ide @mkonicek good to go?

**Comment by ide** [comment]
@facebook-github-bot shipit

**Comment by fangmobile** [comment]
:+1:

**Comment by fangmobile** [comment]
@ mkonicek  anything else I should do?
Sorry for the harassment. We need to have accessibility for them for compliance requirement. Also our Android automation is blocked by this because our Android functional tests use Appium which can not read testId (view tag) so we replying on content descriptions.

**Comment by bestander** [comment]
@fangmobile, sorry, our internal CI got stuck with this one.
A colleagues is taking control over it right now.
Please ping us in a few hours if it does not get resolved, this is a very important PR for the community :+1:

**Comment by fangmobile** [comment]
:+1: 
Thanks guys.

**Pull request closed without merging**